rfcs/2025-03-16-smp-queues.md

```diff
@@ -0,0 +1,163 @@
+
+**Retrieving queue link data**
+
+- the sender uses the public key `k` derived from the private key `pk` included in the link as entity ID to retrieve link data (the server will compute the ID used by the owner as `sha256(k)` and will be able to look it up). This provides the quality that the traffic of the parties has no shared IDs inside TLS. It also means that unlike message queue creation, the ID to retrieve the link data was never sent to the creator, and also is not known to the server in advance (the second part is only an observation, in itself it does not increase security, as server has access to an encrypted blob anyway).
+- note that the sender does not authorize the request to retrieve the blob, as it would not increase security unless a different key is used to authorize, and adding a key would increase link size.
+- server session keys with the sender will be `(sk, spk)`, where `sk` is public key shared with the sender during session handshake, and `spk` is the private key known only to the server.
+- this public key `k` will also be combined with server session key `spk` using `dh(k, spk)` to encrypt the response, so that there is no ciphertext in common in sent and received traffic for these blobs. Correlation ID will be used as a nonce for this encryption.
+- having received the blob, the client can now decrypt it using secret_box with `HKDF(pk)`.
+
+Using the same key as ID for the request, and also to additionally encrypt the response allows to use a single key in the link, without increasing the link size.
+
+## Threat model
+
+See [Short links](./2024-06-21-short-links.md) RFC.
+
+## Syntax for short links
+
+The proposed syntax:
+
+```abnf
+shortConnectionRequest = %s"https://" smpServerHost "/" connReqType "#/" linkKey [ "?" param *( "&" param ) ]
+smpServerHost = <hostname> ; RFC1123, RFC5891
+connReqType = %s"i" / %s"invitation" / %s"c" / %s"contact"
+linkKey = <base64url encoded private key data>
+
+param = fingerprintParam / hostsParam
+fingerprintParam = "f=" fingerprint ; server fingerprint
+hostsParam = "h=" <hostname> *( "," <hostname> ) ; additional hostnames, e.g. onion
+```
+
+To have shorter links fingerpring and additional server hostnames do not need to be specified for preconfigured servers, even if they are disabled - they can be used from the client code.
+
+Example link:
+
+```
+https://smp8.simplex.im/c/#abcdefghij0123456789abcdefghij0123456789abc
+```
+
+For the above to work in the browser the servers should serve server pages.
+
+## Questions
+
+SenderId determination:
+- random, as proposed. Pros: no linkability of observed link to queue. Cons: it can be different from the queue, without the accepting party knowing. Potentially, an attack vector?
+- derived from LinkId, and not changeable: Pros: accepting party can verify. Cons: observed link is linkable to the queue.
```

Comment:
doesn't matter as recipient server is chosen (can be self-hosted) by recipient and can run arbitrary code to run this attack anyway?